414/login signin pages

YuraPetrovskyi opened the pull request:
# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review on my code
- [ ] My changes generate no new warnings or errors
- [ ] New and previously existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description

** Closes #414**  
This PR introduces redesigned Login and Sign Up pages using the newly created Input component and applies various UI improvements for consistency with the Figma v2 design.

It also adds layout refinements such as:

- Footer sticking to the bottom of the screen 
- Removal of the back button in auth layout 
- Background in Storybook stories for better context

### Files changed

- app/(auth)/login/page.tsx: Switched to the new Input component and added validations 
- app/(auth)/signup/page.tsx: Switched to the new Input component, added refugee toggle and validations 
- components/AuthForm.tsx: Refactored logic to support required field errors and updated styles 
- app/(auth)/layout.tsx: Removed back button 
- app/layout.tsx: Ensured footer sticks to the bottom of the screen 
- components/form/Input.stories.tsx: Added container background for realistic preview

### UI changes

Updated Login and Sign Up pages
✅ Improved styling and spacing
✅ Inputs now follow consistent sizing and validation rules
✅ Storybook Input stories show realistic width and background

<p align="center">
  <img src="https://github.com/user-attachments/assets/e5639cc5-1b3c-4655-9017-14fde49886ce" width="250"/>
  <img src="https://github.com/user-attachments/assets/938e9202-970a-45e8-8fc1-6b4896b86474" width="250"/>
  <img src="https://github.com/user-attachments/assets/08ca4f94-5f96-4dab-ac5b-e61d16db28ef" width="250"/>
</p>

### Changes to Documentation

No documentation changes were required.

# Tests

🧪 Only manual testing has been performed at this stage:

- Verified inputs, validation, and password toggle on both login and signup
- Tested error message displays and submit prevention on incomplete forms

YuraPetrovskyi committed:
style(storybook): add background wrapper to better reflect design

relates #414

YuraPetrovskyi committed:
fix: fixed spelling errors

relates #414

YuraPetrovskyi committed:
feat(login/signup): replace form fields with new Input component and toggle for refugee status

relates #414

YuraPetrovskyi committed:
style(layout): ensure footer sticks to bottom of the page

relates #414

YuraPetrovskyi committed:
refactor: remove back button for cleaner design

relates #414

ShaughnAnderson94 reviewed (dismissed):
It all looks good to me identical to the figma as far as I can tell. Its got my approval

YuraPetrovskyi commented:
> It all looks good to me identical to the figma as far as I can tell. Its got my approval

Thanks a lot for taking the time to review it, @ShaughnAnderson94  - really appreciate it! 🙌
I might still spend a bit more time with the code, especially looking into tests and checking why the Supabase Continuous Integration / test (pull_request) job is failing 🤯. Will try to sort that out soon.

YuraPetrovskyi committed:
Merge branch 'dev' of https://github.com/YuraPetrovskyi/Kindly into 414/login-signin-pages

YuraPetrovskyi committed:
Resolve merge conflicts in auth layout, layout and AuthForm

ShaughnAnderson94 commented:
As discussed if the figma doesn't have the back arrow then I think it should be taken out before we close this issue

YuraPetrovskyi committed:
fix: removing the BackButton arrow according to the figma template

relates #414

YuraPetrovskyi commented:
> As discussed if the figma doesn't have the back arrow then I think it should be taken out before we close this issue

Hi @ShaughnAnderson94  
I removed back arrow from our **Log in page** and also removed it from the tests so it wouldn't break. I think it's ready to be merged now🤗

ShaughnAnderson94 reviewed (dismissed):
I still like it

YuraPetrovskyi committed:
Merge branch 'dev' into 414/login-signin-pages
